### PR TITLE
fetching more version based on the amount of pages we can fetch

### PR DESCRIPTION
**Pull request by sagilaufer1992**
the original implementation fetched only the first page. it means newer versions push older ones from the list of results. Therefor this fork TGENV needed a hotfix for older versions. the solution wan’t as pretty as it should have been - we used magic number for the number of pages.
As a solution added functionality for fetching the amount of pages and then fetching those pages too

### CODE REVIEW

File: libexec/tgenv-list-remote [hunk 1]
```diff
@@ -1,5 +1,17 @@
 #!/usr/bin/env bash
-pages=4
+set -e
+
+link_release="https://api.github.com/repos/gruntwork-io/terragrunt/tags?per_page=1000"
+
+# the  response header (--head) contains a link to the last page, contains the last page number and the "last" keyword to help identify
+page_curl=$(curl -s --tlsv1.2 --head "$link_release")
+only_relevant_part=$(echo "$page_curl" | grep -o -E "&page=[0-9]+>; rel=\"last")
```
Comment (eranelbaz) [inline libexec/tgenv-list-remote:8]: `only_relevant_part` relevant to who?
please find better name

**Review by eranelbaz — COMMENTED** [review]
Please update the description,
I don't understand the issue you are trying to solve here...

which original implementation?
since you also set default to 4

File: libexec/tgenv-list-remote (hunk 1 above)
Comment (sagilaufer1992) [reply libexec/tgenv-list-remote]: 9ab31e55

**Comment by sagilaufer1992** [comment]
default is 4 because as for these lines were written that's how many pages there are. In case for any reason the format of the response header changes, I wished to still retrieve extra pages, not just the first, so 4 seems like a reasonable default @eranelbaz

**Review by eranelbaz — APPROVED** [review]

**Pull request merged**